File: Source/Game/Scripts/UI/UIGame.cs
using System;
using System.Collections.Generic;
using FlaxEngine;

namespace Game
{
    public class UIGame : IUIState
    {
        public UIChat Chat;

        public override void OnInitialize()
        {
           
        }

        public override void OnDeinitialize()
        {
            
        }

        public override void OnEnable()
        {
            Screen.CursorLock = CursorLockMode.Locked;
            Screen.CursorVisible = false;
        }

        public override void OnUpdate()
        {
            if(Input.GetKeyUp(KeyboardKeys.T) && !Chat.IsFocused)
            {
                Chat.Focus();
            }
        }

        public override void OnDisable()
        {
            Screen.CursorLock = CursorLockMode.None;
            Screen.CursorVisible = true;
        }

        public override string GetBackState()
        {
            return "None";
        }
    }
}

File: Source/Game/UI/UIGame.cpp
#include "UIGame.h"

UIGame::UIGame(const SpawnParams& params)
    : IUIState(params)
{
    _tickUpdate = true;
}

void UIGame::OnInitialize()
{
    Screen::SetCursorLock(CursorLockMode::Locked);
    Screen::SetCursorVisible(false);
    IUIState::OnInitialize();
}

void UIGame::OnDeinitialize()
{
    Screen::SetCursorLock(CursorLockMode::None);
    Screen::SetCursorVisible(true);
    IUIState::OnDeinitialize();
}

void UIGame::OnUpdate()
{
    if (Input::GetKeyUp(KeyboardKeys::T) && !Chat->IsFocused())
    {
        Chat->Focus();
    }
}

String UIGame::GetBackState()
{
    return TEXT("None");
}
